feat(frontend): add related content to letters and add tools and letters to related contents

UnbearableBear opened the pull request:
closes #1519 closes #1308 
<img width="1495" alt="Screenshot 2019-10-23 at 14 50 30" src="https://user-images.githubusercontent.com/705453/67396633-368c8380-f5a8-11e9-9396-193fecb24385.png">

Commit:
feat(frontend): add related content to letters and add tools and letters to related contents

Commit:
test(frontend): update snapshots

UnbearableBear commented:
@nkrmr Désolé, pour adapter aux outils et lettres j'ai du revoir un peu le design !

lionelB commented:
@UnbearableBear  on peut avoir la liste pour les documents et une carte pour le premier outil ? et que l'outils soit au debut ?

lionelB commented on packages/code-du-travail-frontend/src/common/Answer.js:
```diff
@@ -84,29 +109,62 @@ function Answer({
+                      as={`/${getRouteBySource(link.source)}/${link.slug}`}
+                      passHref
+                    >
+                      <LargeLink variant="light">
```
Can we keept the previous style for Article and add Tile block under ?

UnbearableBear replied:
I followed the specs given by clement concerning the order but I'm going to make some proposals and see what we decide

lionelB reviewed (changes requested):
It would be nice a to add a snapshot on modele-courriers page taht check we have some related items

UnbearableBear commented:
@lionelB modele-courriers page uses `Answer` but you're right, I'll add a test to ensure related items are displayed when `Answer` is given the correct data

UnbearableBear commented:
@ClementChapalain Une préférence ? (Sur l'ordre aussi) 

<img width="1495" alt="Screenshot 2019-10-23 at 14 50 30" src="https://user-images.githubusercontent.com/705453/67421888-aebb6f00-f5d1-11e9-9a8d-616ed79d9e94.png">
<img width="1306" alt="Screenshot 2019-10-23 at 20 12 05" src="https://user-images.githubusercontent.com/705453/67421890-aebb6f00-f5d1-11e9-896d-b0bcded48b49.png">
<img width="1284" alt="Screenshot 2019-10-23 at 20 39 47" src="https://user-images.githubusercontent.com/705453/67423853-49697d00-f5d5-11e9-953e-717983749483.png">

ClementChapalain commented:
celui là ! si tu peux d'ailleurs en profiter pour virer la ligne à gauche des articles ce serait top je pense :)

![image](https://user-images.githubusercontent.com/5337489/67464517-bd426e80-f643-11e9-892d-c0c74b62ed4d.png)

Commit:
feat(ui): add size prop to tile and buttons

Commit:
test(ui): update snapshots

Commit:
test(frontend): add tests and update snapshots

Commit:
refactor(chore); update lockfile

UnbearableBear commented:
Et voici le résultat final !
<img width="1283" alt="Screenshot 2019-10-24 at 14 39 06" src="https://user-images.githubusercontent.com/705453/67486318-1f16ce80-f66c-11e9-8dbe-6aa1024d2b44.png">

Commit:
fix(frontendt): typo

lionelB commented on packages/code-du-travail-frontend/src/common/__tests__/Answer.test.js:
```diff
@@ -1,10 +1,11 @@
 
 describe("<Answer />", () => {
-  it("should render", () => {
+  it("renders", () => {
```
all the tests are using it("should")  can we keep consistency here ?

lionelB commented on packages/react-ui/src/Grid/index.js:
```diff
@@ -17,15 +18,15 @@ export const Grid = styled.ul`
+  display: flex;
   flex-grow: 0;
   flex-shrink: 1;
+  width: calc(100% / 4 - 2 * ${spacing.small});
```
it's not clear what was wrong with flex-basis that width solves  ?

UnbearableBear replied:
It solves nor changes absolutely nothing, I was tinkering with IE11 and forgot to put the old value back !

lionelB reviewed (commented):
It seems that you have added som ui component too but I don't see them, can you add some screenshot of what was added and what usage ?

UnbearableBear commented:
I did not add any component, however, I realized that the tile would always take 100% of the container width which is not always wanted so I added a `display: inline-flex` instead of a `display: flex` and made changes accordingly. I also mirrored the `Tile` props to the `LargeLink` prop concerning the `variant` prop. Finally I added a `size` prop to both the `Button` and the `Tile` component which can now have slightly smaller versions (used in the linked contents)

UnbearableBear commented:
<img width="450" alt="Screenshot 2019-10-24 at 19 07 47" src="https://user-images.githubusercontent.com/705453/67508664-b7737a00-f691-11e9-8d63-3aaa7fac25ff.png">
<img width="719" alt="Screenshot 2019-10-24 at 19 07 37" src="https://user-images.githubusercontent.com/705453/67508665-b7737a00-f691-11e9-8bf8-c7e976c78608.png">
<img width="653" alt="Screenshot 2019-10-24 at 19 07 29" src="https://user-images.githubusercontent.com/705453/67508666-b80c1080-f691-11e9-8f31-bc23d40d2e7f.png">
